snap/local/start-mysql-router-exporter.sh

```diff
@@ -30,16 +32,32 @@ if [ -n "$SNAP" ]; then
         exit 1
     fi
 
-    # For security measures, daemons should not be run as sudo.
-    # Execute mysqlrouter-exporter as the non-sudo user: snap-daemon.
-    exec "$SNAP"/usr/bin/setpriv \
-        --clear-groups \
-        --reuid snap_daemon \
-        --regid snap_daemon -- \
-        env MYSQLROUTER_EXPORTER_URL="${MYSQLROUTER_EXPORTER_URL}" \
-        MYSQLROUTER_EXPORTER_USER="${MYSQLROUTER_EXPORTER_USER}" \
-        MYSQLROUTER_EXPORTER_PASS="${MYSQLROUTER_EXPORTER_PASS}" \
-        "$EXPORTER_PATH" "${EXPORTER_OPTS[@]}" "$TLS_OPTS"
+    if [[ -n "$MYSQLROUTER_TLS_CACERT_PATH" && -n "$MYSQLROUTER_TLS_CERT_PATH" && -n "$MYSQLROUTER_TLS_KEY_PATH" ]]; then
+        # For security measures, daemons should not be run as sudo.
+        # Execute mysqlrouter-exporter as the non-sudo user: snap-daemon.
+        exec "$SNAP"/usr/bin/setpriv \
+            --clear-groups \
+            --reuid snap_daemon \
+            --regid snap_daemon -- \
+            env MYSQLROUTER_EXPORTER_URL="${MYSQLROUTER_EXPORTER_URL}" \
+            MYSQLROUTER_EXPORTER_USER="${MYSQLROUTER_EXPORTER_USER}" \
+            MYSQLROUTER_EXPORTER_PASS="${MYSQLROUTER_EXPORTER_PASS}" \
+            MYSQLROUTER_TLS_CACERT_PATH="${MYSQLROUTER_TLS_CACERT_PATH}" \
+            MYSQLROUTER_TLS_CERT_PATH="${MYSQLROUTER_TLS_CERT_PATH}" \
+            MYSQLROUTER_TLS_KEY_PATH="${MYSQLROUTER_TLS_KEY_PATH}" \
+            "$EXPORTER_PATH" "${EXPORTER_OPTS[@]}"
+    else
+        # For security measures, daemons should not be run as sudo.
+        # Execute mysqlrouter-exporter as the non-sudo user: snap-daemon.
+        exec "$SNAP"/usr/bin/setpriv \
+            --clear-groups \
+            --reuid snap_daemon \
+            --regid snap_daemon -- \
+            env MYSQLROUTER_EXPORTER_URL="${MYSQLROUTER_EXPORTER_URL}" \
+            MYSQLROUTER_EXPORTER_USER="${MYSQLROUTER_EXPORTER_USER}" \
+            MYSQLROUTER_EXPORTER_PASS="${MYSQLROUTER_EXPORTER_PASS}" \
+            "$EXPORTER_PATH" "${EXPORTER_OPTS[@]}" --skip-tls-verify
+    fi
```

Comment:
Not tested, but IMHO easier to read/support in the future:

```suggestion
    SETPRIV_OPT=(
        "--clear-groups"
        "--reuid snap_daemon"
        "--regid snap_daemon"
    )

    EXPORTER_ENV=(
        "MYSQLROUTER_EXPORTER_URL='${MYSQLROUTER_EXPORTER_URL}'"
        "MYSQLROUTER_EXPORTER_USER='${MYSQLROUTER_EXPORTER_USER}'"
        "MYSQLROUTER_EXPORTER_PASS='${MYSQLROUTER_EXPORTER_PASS}'"
    )
    
    if [[ -n "${MYSQLROUTER_TLS_CACERT_PATH}" && \
          -n "${MYSQLROUTER_TLS_CERT_PATH}" && \
          -n "${MYSQLROUTER_TLS_KEY_PATH}" ]]; then
        EXPORTER_ENV+=("MYSQLROUTER_TLS_CACERT_PATH='${MYSQLROUTER_TLS_CACERT_PATH}'")
        EXPORTER_ENV+=("MYSQLROUTER_TLS_CERT_PATH='${MYSQLROUTER_TLS_CERT_PATH}'")
        EXPORTER_ENV+=("MYSQLROUTER_TLS_KEY_PATH='${MYSQLROUTER_TLS_KEY_PATH}'")
    else
        EXPORTER_OPTS+=("--skip-tls-verify")
    fi
    
    # For security measures, daemons should not be run as sudo.
    # Execute mysqlrouter-exporter as the non-sudo user: snap-daemon.
    exec "${SNAP}"/usr/bin/setpriv "${SETPRIV_OPT[@]}" -- \
        env  "${EXPORTER_ENV[@]}" "${EXPORTER_PATH}" "${EXPORTER_OPTS[@]}"
```

Reply:
in the spirit of all the readability feedback, refactored in fa4e9a8c4b064b02274bd4bfa528966a8e690469